src/UrlBuilder.php

```diff
@@ -56,16 +56,15 @@ public function generateTransfer(string $recipient, array $options = []): string
         return $this->generateUrl($options);
     }
 
-    public function generateVote(string $subject): string
+    public function generateVote(string $validatorPublicKey, ?string $username = null): string
     {
         $options = [
-            'method' => Methods::Vote->name(),
+            'method'    => Methods::Vote->name(),
+            'publicKey' => $validatorPublicKey,
         ];
 
-        if (strlen($subject) === 66) {
-            $options['publicKey'] = $subject;
-        } else {
-            $options['delegate'] = $subject;
+        if ($username) {
+            $options['username'] = $username;
```

Comment:
```suggestion
            $options['delegate'] = $username;
```
should remain as delegate (or validator, but that's better in a separate branch)

Reply:
I have no idea why I changed it 🤦